hooks/shapediver/viewer/interaction/useNodeInteractionData.ts

```diff
@@ -0,0 +1,128 @@
+	 */
+	outputNode: ITreeNode | undefined
+} {
+	/**
+	 * Callback function to add interaction data to the nodes of the output.
+	 * 
+	 * @param node
+	 */
+	const callback = useCallback((node?: ITreeNode) => {
+		if(!node) return;
+		/**
+		 * Check if the node matches the pattern and add interaction data if it does.
+		 * 
+		 * For each pattern, the function will check if the node name matches the pattern.
+		 * If the node name matches the pattern, the function will check the children of the node.
+		 * Only if the node name matches the last pattern, the interaction data will be added.
+		 * 
+		 * @param node The node to check.
+		 * @param pattern The pattern to check.
+		 * @param count The current count of the pattern.
+		 * @param result The result array. 
+		 */
+		const checkNode = (node: ITreeNode, pattern: string[], count: number, result: ITreeNode[] = []): void => {
+			if(new RegExp(`^${pattern[count]}$`).test(node.name)) {
+				if(count === pattern.length - 1) result.push(node);
+
+				for(const child of node.children) {
+					checkNode(child, pattern, count === pattern.length - 1 ? 0 : count + 1, result);
+				}
+			} else {
+				for(const child of node.children) {
+					checkNode(child, pattern, 0, result);
+				}
+			}
+		};
+
+		// if there are patterns, begin the check
+		if (patterns && interactionSettings) {
+			for (const pattern of patterns) {
+				const nodes: ITreeNode[] = [];
+				checkNode(node, pattern, 0, nodes);
+				nodes.forEach(node => {
+					// remove the interaction data if it already exists
+					if (nodesWithInteractionData[node.id]) {
+						nodesWithInteractionData[node.id].node.removeData(nodesWithInteractionData[node.id].data);
+						delete nodesWithInteractionData[node.id];
+					}
+
+					// add the interaction data to the node
+					const interactionData = new InteractionData({
+						select: interactionSettings.select,
+						hover: interactionSettings.hover,
+						drag: interactionSettings.drag
+					});
+					node.addData(interactionData);
+					node.updateVersion();
+					nodesWithInteractionData[node.id] = { node, data: interactionData };
+				});
+			}
+		}
+
+		// call the additional update callback
+		if(additionalUpdateCallback)
+			additionalUpdateCallback(node);
+	}, [patterns, interactionSettings, additionalUpdateCallback] );
+	
+
+	// define the node update callback
+	const { outputApi, outputNode } = useOutputNode(sessionId, outputIdOrName, callback);
+
+	useEffect(() => {
+		// call the callback with the output node
+		callback(outputNode);
+
+		return () => {
+			// remove the interaction data from the nodes
+			for(const id in nodesWithInteractionData) {
+				const { node, data } = nodesWithInteractionData[id];
+				node.removeData(data);
+				node.updateVersion();
+			}
+			// clear the dictionary
+			Object.keys(nodesWithInteractionData).forEach(key => delete nodesWithInteractionData[key]);
+		};
+		
+	}, [interactionSettings, patterns, additionalUpdateCallback]);
```

Comment:
@MajorMeerkatThe3rd let's also add `callback` to the dependencies (although it's covered by the other dependencies, but who knows what we change in the future)